[IMP] product_extended_variants: Now cron job is splited in several parts.

nhomar opened the pull request:
- [x] Allow thousands of BOM being computed without wait 2 days for such deal the cronjob must create as many as cron jobs asked to be recomputed in big databases.
- [x] Disabled by default the cron job in order to allow overwrite process.
- [ ] This way is too complex to test, due to this we must develop unit tests in order to avoid introduce regressions in a future.
- [ ] Refactor on separate methods, but only when unit tests are done. it is working right now.
- [ ] Clean up unused methods.
- [ ] Make the numbers a little bit smarter, it means:
  - [ ] What happen if not too much products?.
  - [ ] What happen if len(Product_ids) too little?.
  - [ ] What happen if no product_ids at all?
  - [ ] What happen if product_template?

luistorresm commented:
@nhomar 

Why not use new api in new method execute_cron_splited?

Regards

luistorresm commented:
@nhomar 

I added tests in this PR, can you check this please?

Regards

luistorresm commented:
@nhomar 

Can you please check the tests in this PR, to continue with the TO-DO?

Regards

luistorresm commented:
@nhomar 

Alguna novedad con la revisión de esto?

Saludos

luistorresm commented:
@nhomar 

Adjunto los coverage de esta PR para los py de product_extended_variants.
[Coverage for _root_addons-vauxoo_product_extended_variants_model_template.pdf](https://github.com/Vauxoo/addons-vauxoo/files/366570/Coverage.for._root_addons-vauxoo_product_extended_variants_model_template.pdf)
[Coverage for _root_addons-vauxoo_product_extended_variants_wizard_wizard_price.pdf](https://github.com/Vauxoo/addons-vauxoo/files/366572/Coverage.for._root_addons-vauxoo_product_extended_variants_wizard_wizard_price.pdf)

Para continuar con esta PR debe estar todo en 100% o lo revisas primero?

Saludos

luistorresm commented:
@nhomar 

Esta semana se comprometió la entrega de este desarrollo.

Adjunto nuevamente los resultados del coverage:

[Coverage for _root_addons-vauxoo_product_extended_variants_model_template.pdf](https://github.com/Vauxoo/addons-vauxoo/files/439447/Coverage.for._root_addons-vauxoo_product_extended_variants_model_template.pdf)
[Coverage for _root_addons-vauxoo_product_extended_variants_wizard_wizard_price.pdf](https://github.com/Vauxoo/addons-vauxoo/files/439448/Coverage.for._root_addons-vauxoo_product_extended_variants_wizard_wizard_price.pdf)

![vauxoo lodigroup build 2405 coveralls test coverage history statistics](https://cloud.githubusercontent.com/assets/7606656/18008445/bb785128-6b6d-11e6-94c7-b6694fda48ec.png)

Una de las cosas que me pediste era que antes de limpiar el código, me deberías autorizar que los tests están correctos.

"¿Me echas la mano?" Para poder hacer la limpieza y poder entregar esto?

Saludos

luistorresm committed:
[IMP] product_extended_variants: Now cron job is splited in several
parts in order allow thousands of BOM being computed without wait
2 days for such deal.

Disabled by default the cron job in order to allow overwrite process.

- Improve code in methods
- Fixed that update all products, when your bom parent is a standard
  price, although the bom line product are different, then now is
  updated the product, to get the correct standard price in recursive
  actions, but after of this process is reverted the cost in products
  that cost method is different to standard, by this reason was added
  improve that only create account movement when the product cost
  method is "standard".
- Added test with a recursive method, to verify that only is updated the
  product cost when is standard, and that the price is correct.
- Added needed test with other user cases.
- Added new method execute_cron_splited, that now is called from the
  cron that generate the cron splited.

[FIX] product_extended_segmentation: Fixed that not update price in
products with cost method != standard.

Added tests to verify that this validation is correct.

luistorresm committed:
[IMP] product_extended_variants: Completed README file

luistorresm committed:
[IMP] product_extended_variants: Fixed lint errors

luistorresm committed:
[REF] product_extended_segmentation: Fixed lint errors

luistorresm committed:
[IMP] product_extended_variants: Fixed reference in test

luistorresm commented:
@nhomar 

Could you check and merge this PR?

Regards

luistorresm commented:
lint error:
![job 3523 1 vauxoo addons vauxoo travis ci](https://cloud.githubusercontent.com/assets/7606656/19084245/61648abc-8a2b-11e6-9705-dd778b4cd0e0.png)

luistorresm commented:
@josemoralesp 

Could you merge this PR? The customer was approve.

Regards
